### PR TITLE
Encapsulate `CreateTable`, `CreateIndex` into specific structs

### PR DESCRIPTION
**Pull request by philipcristiano**
cc @alamb, @tisonkun 

Continue moving structs as part of #1204 

The `CreateTableBuilder` could maybe be combined into the `CreateTable`. For the purposes of minimizing breaking changes I did not attempt to merge them.

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
Somehow "trivial" changes :D

LGTM.

**Comment by philipcristiano** [comment]
I have a branch for `CreateIndex` as well, should I add it to this PR or make a new one?

**Comment by tisonkun** [comment]
@philipcristiano either should work. If it's likely to conflict, I'd prefer to add on this one. As I suppose it would be not quite complex.

**Comment by philipcristiano** [comment]
Added to this PR! 

`cargo test` and `cargo test --all-features` passes for me locally. Once workflows are approved I can see if there was anything else missing that I haven't covered.

**Comment by alamb** [comment]
I started the CI checks

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @philipcristiano  and @tisonkun 

cc @jmhain  and @iffyio -- what do you think?

File: src/ast/dml.rs
Comment (jmhain) [inline src/ast/dml.rs:29]:
```suggestion
/// CREATE INDEX statement.
```

**Review by jmhain — APPROVED** [review]

**Review by iffyio — APPROVED** [review]
LGTM!

**Review by tisonkun — COMMENTED** [review]
One suggestion: We may implement `Display` on this struct so that it's more smooth to use downstream.

Currently, IIUC we use the `pub` fields and implement the top level `Display` manually.

**Comment by philipcristiano** [comment]
I moved the CreateTable display implementation from Statement to the struct.

**Comment by philipcristiano** [comment]
Is there anything else that I should do before this could be merged?

**Comment by alamb** [comment]
Hi @philipcristiano -- sorry -- no you don't need to do anything else. I am somewhat behind as I have been struggling to update DataFusion with the latest sqlparser release and thus haven't had bandwidth to merge more changes to SQL parser that have significant breaking changes

I think I am mostly unstuck now and will merge this one shortly

**Comment by philipcristiano** [comment]
Awesome! Thank you for your help on this issue and your work on this project!

**Pull request merged**
